### PR TITLE
Improve Windows Installation instructions

### PR DESCRIPTION
**Pull request by peci1**
# 🎉 New feature

## Summary
Going through the Windows install instructions for Harmonic, these are a few changes that will hopefully enable more users to finish the installation.

## Test it
Try installing it :-D

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: harmonic/install_windows_src.md [hunk 1]
```diff
@@ -58,6 +67,8 @@ You will still be able to use `TPE` as a physics engine
   libzip qt pybind11 --channel conda-forge
   ```
 
+  This can take tens of minutes.
```
Comment (traversaro) [inline harmonic/install_windows_src.md:70]: Not sure if we want either to suggest mamba or at least to enable conda's mamba based backend (see https://www.anaconda.com/blog/a-faster-conda-for-a-growing-community). I guess conda was originally used here as it works out of the box on Anaconda/miniconda while mamba only works out of the box in miniforge, while for mamba you need to document how to install it. However nowadays conda's mamba backend should work out of the box also in miniconda.

File: harmonic/install_windows_src.md [hunk 2]
```diff
@@ -117,7 +125,7 @@ If there are no errors, all the binaries should be ready to use.
 
 ## Using the workspace
 
-The workspace needs to be sourced every time a new terminal is used.
+The workspace needs to be sourced every time a new terminal is used (and Conda activated before that).
```
Comment (traversaro) [inline harmonic/install_windows_src.md:128]: Cobra activated -> conda environment activated

**Review by traversaro — APPROVED** [review]
Minor comments, thanks for working on this!

File: harmonic/install_windows_src.md [hunk 3]
```diff
@@ -4,13 +4,14 @@ WARNING: Current Windows support is experimental.
 
 # Source Installation on Windows 10 or 11
 
-Command line tools, DART physics engine, and GUI capabilities are
+DART physics engine and GUI capabilities are
 not currently supported in Windows. These functionalities correspond to the currently
-building packages `gz-tools`, `gz-physics`, and `gz-gui`, respectively.
+building packages `gz-physics` and `gz-gui`, respectively. The packages will build,
+but you can expect runtime failures when using their functionalities.
```
[the remark this answers is not in the document]
Comment (peci1) [reply harmonic/install_windows_src.md:10]: You're right, dartsim works out of the box! And rendering works (at least for a while) with ogre. I'll update the PR.

File: harmonic/install_windows_src.md (hunk 1 above)
Comment (peci1) [reply harmonic/install_windows_src.md:70]: Sure, I always wondered why there's conda and mamba =)

File: harmonic/install_windows_src.md [hunk 4]
```diff
@@ -21,14 +20,17 @@ You should be able to use `TPE` as a physics engine without any issues
...
-2. Install [Visual Studio 2019](https://visualstudio.microsoft.com/downloads/).
+2. Install [Visual Studio 2019 or 2022](https://visualstudio.microsoft.com/downloads/).
    The Community version is free for students, open-source developers, and personal
    development. Check "Desktop development with C++" in the Workloads tab,
    check "MFC and ATL support", and uncheck "C++ Cmake Tools." We will install
-   cmake via Conda.
+   cmake via Conda. All other checkboxes can be left unchecked.
+   It is recommended to choose the Visual Studio version based on
+   the supported versions for your ROS distribution if you plan to use Gazebo with
+   ROS.
```
Comment (traversaro) [inline harmonic/install_windows_src.md:30]: I am not sure if we want to mention this. With all probability, this binaries of Gazebo are not ABI compatible with any ROS on Windows distribution except for RoboStack, and even if it is, it is probably just by luck and this could be break at any time due to ABI changes in conda-forge packages.

Comment (peci1) [reply harmonic/install_windows_src.md:30]: Oh, I wasn't meaning any kind of ABI compatibility. This was much more meant as a newbie advice so that they don't need to install two versions of VS, one for Gazebo and one for ROS.

Comment (traversaro) [reply harmonic/install_windows_src.md:30]: I am afraid that most people will read "if you plan to use Gazebo with ROS." as using [`ros_gz`](https://github.com/gazebosim/ros_gz)

Comment (peci1) [reply harmonic/install_windows_src.md:30]: Exactly. Is that a problem?

Comment (traversaro) [reply harmonic/install_windows_src.md:30]: Unless I am missing something, that sentence seems to be hint that in general Windows ROS distributions will work with Gazebo installed from source via conda-forge dependency, i.e. reading that I would expect that following the official Windows install instructions (i.e. https://docs.ros.org/en/iron/Installation/Windows-Install-Binary.html) I can install ros_gz (perhaps from source) and everything could work. In reality, the two distributions may install different version (with possibly different ABI) of each shared package (for example `yaml-cpp`) that can lead to runtime segfaults.

Comment (peci1) [reply harmonic/install_windows_src.md:30]: Ahh, ROS decided to go with chocolatey... So this could really be a problem. So do you suggest completely removing any mention of ROS? Or better saying that this guide should not be used if ros_gz is to be built?

Comment (traversaro) [reply harmonic/install_windows_src.md:30]: The easiest solution is just not to mention ROS, so I would go at least for that.  The only reasonable way that I see for getting ros_gz to work with gz installed with conda-forge dependencies is either to build all of ROS2 with conda-forge dependencies, or use RoboStack (but there ros_gz is not working at the moment, see https://github.com/RoboStack/ros-humble/issues/68), so the situation is not easy to explain.

Comment (peci1) [reply harmonic/install_windows_src.md:30]: On the other hand, if it's not mentioned, people might just try it, losing lots of time and being confused. What about changing the wording to:

> It is recommended to choose the Visual Studio version based on the supported versions for your ROS 2 distribution if you plan to use Gazebo with ROS 2. However, be aware that the default Binary ROS 2 installation guide uses vcpkg for installation of the dependencies, and that is not compatible with conda-installed packages from this guide. To have Gazebo and ROS 2 both working on the same system, you need to choose either vcpkg, or conda (or source builds) and use it for the dependencies of both Gazebo and ROS 2. Unfortunately, to this date, there is no guide for this approach available.

Comment (traversaro) [reply harmonic/install_windows_src.md:30]: I like it, thanks! 

Regarding the sentence:

> be aware that the default Binary ROS 2 installation guide uses vcpkg for installation of the dependencies, and that is not compatible with conda-installed packages from this guide. 

I would rephrase it as:

> be aware that the default Binary ROS 2 installation guide does not use conda-forge dependencies for installation of the dependencies does, and that is not compatible with conda-forge packages from this guide. 

For these reasons:
* the installation https://docs.ros.org/en/iron/Installation/Windows-Install-Binary.html does not use vcpkg for dependencies, but just a mix of manually compiled libraries, some delivered via Chocolatey and some that need to be installed manually by downloading .zip files. 
* (this is mostly a nitpick, feel free to ignore) what ensure that the packages that we install are abi-compatible is not that they are installed via the conda package manager, but rather that they are installed via the conda-forge "distribution".

Comment (mabelzhang) [reply harmonic/install_windows_src.md:30]: So, I agree that we shouldn't mention ROS on this page. I mean, at all. There is a page dedicated to ROS and Gazebo interoperability, and installing them together. Anything ROS should go there https://gazebosim.org/docs/harmonic/ros_installation

The Gazebo installation pages should be Gazebo-only. Otherwise there will be no end of mixing ROS instructions in the Gazebo website. That would just be very unmaintainable.

Comment (peci1) [reply harmonic/install_windows_src.md:30]: Ok, Mabel's argument makes sense. I'll remove the ROS mentions.

**Comment by peci1** [comment]
All comments should be resolved now.

File: harmonic/install_windows_src.md [hunk 5]
```diff
@@ -132,6 +142,34 @@ call install\setup.bat
 This is the end of the source install instructions; head back to the [Getting started](/docs/all/getstarted)
 page to start using Gazebo!
 
+> [!WARNING]
+> As Gazebo GUI is not yet working, running `gz sim` will not work. You can run only the server with
+
+```cmd
+gz sim -s -v
+```
```
Comment (mabelzhang) [inline harmonic/install_windows_src.md:150]: Same as above.
I tried the `<div>` CSS here to get a nice yellow warning bar, but once we use `<div>`, we don't get the Markdown parsing that makes the code font face. So it looks better with just simple Markdown.
Code block needs to be indented too. I've tested this on the rendered page.
```suggestion
> **WARNING**
> As Gazebo GUI is not yet working, running `gz sim` will not work. You can run only the server with
> ```cmd
> gz sim -s -v
> ```
```

File: harmonic/install_windows_src.md [hunk 6]
```diff
@@ -132,6 +142,34 @@ call install\setup.bat
...
+> [!WARNING]
+> If you username contains spaces (which is quite common on Windows), you will probably get errors
+  saying `Invalid partition name [Computer:My User With Spaces]`. Fix this by changing `GZ_PARTITION`
+  to something else:
+
+```cmd
+set GZ_PARTITION=test
+```
+
+Remember to set the same partition in all other consoles.
```
Comment (mabelzhang) [inline harmonic/install_windows_src.md:161]: Same as above.
And all the lines need to be indented.
With the suggested change, result looks like this:
Screenshot from 2023-09-15 21-20-16

```suggestion
> **WARNING**
> If your username contains spaces (which is quite common on Windows), you will probably get errors
>  saying `Invalid partition name [Computer:My User With Spaces]`. Fix this by changing `GZ_PARTITION`
>  to something else:
> ```cmd
> set GZ_PARTITION=test
> ```
> Remember to set the same partition in all other consoles.
```

Comment (mabelzhang) [reply harmonic/install_windows_src.md:161]: I would suggest to use a word other than `WARNING` though, because that's rather severe, and to differentiate from the yellow warning bar up top.
It sounds more like a tip or a troubleshoot to me. What makes these WARNINGs different from a NOTE? Can they just be NOTEs?

File: harmonic/install_windows_src.md [hunk 7]
```diff
@@ -20,62 +20,74 @@ You will still be able to use `TPE` as a physics engine
...
+   You can use `conda info --envs` to see all of your environments.
+
+   To speed up conda installations, also set the following to use libmamba solver.
+   Older conda installations may need to do [additional steps](https://www.anaconda.com/blog/a-faster-conda-for-a-growing-community).
+   ```bash
+   conda config --set solver libmamba
+   ```
+   To remove an environment, use `conda remove --all --name <env_name>`.
+
+   > [!NOTE]
```
Comment (mabelzhang) [inline harmonic/install_windows_src.md:55]: Same as above
```suggestion
   > **NOTE**
```

File: harmonic/install_windows_src.md [hunk 8]
```diff
@@ -20,62 +20,74 @@ You will still be able to use `TPE` as a physics engine
...
-  libzip qt pybind11 --channel conda-forge
-  ```
+   ```bash
+   conda install cmake git vcstool curl pkg-config ^
+   colcon-common-extensions dartsim eigen freeimage gdal gts ^
+   glib dlfcn-win32 ffmpeg ruby tinyxml2 tinyxml ^
+   libprotobuf urdfdom zeromq cppzmq ogre=1.10 ogre-next jsoncpp ^
+   libzip qt pybind11 --channel conda-forge
+   ```
+  This can take tens of minutes (or less when using libmamba solver).
```
Comment (mabelzhang) [inline harmonic/install_windows_src.md:71]:
```suggestion
   This can take tens of minutes (or less when using libmamba solver).
```

File: harmonic/install_windows_src.md [hunk 9]
```diff
@@ -4,14 +4,14 @@ WARNING: Current Windows support is experimental.
...
+OGRE2 rendering capabilities are not currently supported in Windows, and Gazebo GUI
+works in a limited fashion. These functionalities correspond to the currently
+building packages `gz-rendering` and `gz-sim`, respectively. The packages will build,
+but you can expect runtime failures when using their functionalities.
 
-**Note**
-
-You will still be able to use `TPE` as a physics engine
-(see [here](https://gazebosim.org/api/physics/2.2/physicsplugin.html) for more information on `TPE`).
+> [!NOTE]
```
Comment (mabelzhang) [inline harmonic/install_windows_src.md:12]: This `[!NOTE]` is a GitHub Markdown format that will not work on the rendered webpage. The `docs` repo is meant to be deployed to the Gazebo website, so the formatting has to target the website, not GitHub. You can see the rendered page here https://gazebosim.org/docs/harmonic/install_windows_src

For fancy stuff like warning bars (see the first line in this file), we have to add CSS to the website frontend for them to work. For example, the yellow warning bar format was added here: https://github.com/gazebo-web/gazebosim-web-frontend/pull/12

For now, we need to use the original format here. Otherwise it just looks like this when I test the PR locally:
Screenshot from 2023-09-15 20-16-04
```suggestion
> **NOTE**
```

**Comment by mabelzhang** [comment]
This PR should be reviewed by someone who's familiar with Windows. There are bits about packaging and rendering. Maybe @j-rivero and/or @iche033?

**Comment by peci1** [comment]
Thanks for the formatting improvements. I turned all WARNINGs into NOTEs.
